core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1634,71 +1674,229 @@ boolean resumeSyncForDocument(
    * @param namespace the namespace to put the document in.
    * @param document  the document to insert.
    */
-  public void insertOneAndSync(
-      final MongoNamespace namespace,
-      final BsonDocument document
-  ) {
+  void insertOneAndSync(final MongoNamespace namespace, final BsonDocument document) {
     getLocalCollection(namespace).insertOne(document);
-    final ChangeEvent<BsonDocument> event =
-        changeEventForLocalInsert(namespace, document, true);
+    final BsonValue documentId = BsonUtils.getDocumentId(document);
+    final ChangeEvent<BsonDocument> event = changeEventForLocalInsert(namespace, document, true);
     final CoreDocumentSynchronizationConfig config = syncConfig.addSynchronizedDocument(
         namespace,
-        BsonUtils.getDocumentId(document)
+        documentId
     );
     config.setSomePendingWrites(logicalT, event);
-    final BsonValue documentId = BsonUtils.getDocumentId(document);
     triggerListeningToNamespace(namespace);
     emitEvent(documentId, event);
   }
 
   /**
-   * Updates a single synchronized document by its given id with the given update specifiers.
-   * No update will occur if the _id is not being synchronized.
+   * Inserts one or more documents.
    *
-   * @param namespace  the namespace where the document lives.
-   * @param documentId the _id of the document.
-   * @param update     the update modifiers.
-   * @return the result of the update.
+   * @param documents the documents to insert
    */
-  public UpdateResult updateOneById(
-      final MongoNamespace namespace,
-      final BsonValue documentId,
-      final BsonDocument update
-  ) {
-    // TODO: lock down id
-    final CoreDocumentSynchronizationConfig config =
-        syncConfig.getSynchronizedDocument(namespace, documentId);
-    if (config == null) {
-      return UpdateResult.acknowledged(0, 0L, null);
+  void insertManyAndSync(final MongoNamespace namespace,
+                         final List<BsonDocument> documents) {
+    getLocalCollection(namespace).insertMany(documents);
```

Comment:
[alternatively] instead of generating IDs client-side, and depending on `BsonUtils.getDocumentId` not failing we could look at the result of the local `insertMany` (and `insertOne` in `insertOneAndSync`), and use the inserted ids there to update the doc config and emit the event

Reply:
Should note that there are no results from local inserts.

Reply:
The local client does not return a result.

Reply:
my mistake, I assumed it returned a an insertedId like Stitch